src/colony/lua_tm.c

```diff
@@ -427,9 +432,13 @@ static int l_tm_ssl_read (lua_State* L)
   uint8_t buf[20000];
   size_t buf_len = sizeof(buf);
   int err = tm_ssl_read(session, buf, &buf_len);
-  (void) err;
+  if (err < 0) {
```

Comment:
No, the underlying `ssl_read` call just returns -1 on error (that's maybe where the negative idea comes from), but the actual error is set via the `errno` quasi-global variable and that will usually be positive.

So I would say `if (err)` [or `if (err != 0)` if that convention is preferred] should be the check for an error status.